torchgeo/datasets/ssl4eo.py

```diff
@@ -157,6 +221,62 @@ def __len__(self) -> int:
         """
         return len(self.scenes)
 
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        path = os.path.join(self.subdir, "*", "*", "all_bands.tif")
```

Comment:
This is likely going to be very slow and must be executed every time you instantiate the dataset. Thoughts on how to make this faster?

Reply:
It's worse than I thought. 21 minutes to glob everything. There are a few ways to speed this up. We could use `00000*` instead of `*` so we're only checking for a subset of directories. This should work for all datasets. Alternatively, we could change the indices to be 1–250K instead of the original sample indices. This would make a parallel corpus across sensors more difficult, but we would no longer need to use os.listdir or glob.glob. Note that the sorted os.listdir in `__init__` is almost instantaneous so that isn't of concern.